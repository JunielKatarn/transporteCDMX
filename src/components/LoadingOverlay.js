import React, { memo } from 'react'
import { StyleSheet, View } from 'react-native'
import Loader from './Loader'
import theme, { sharedStyles } from '../styles'

const LoadingOverlay = ({ showOverlay, loaderColor }) => {
  if (showOverlay) {
    return (
      <View style={[styles.overlay, theme.colors.bgBlackOpacity, sharedStyles.zIndex5]}>
        <Loader color={loaderColor} textStyle={theme.colors.colorPrimary} visible />
      </View>
    )
  }
  return null
}

const styles = StyleSheet.create({
  overlay: {
    ...sharedStyles.centerChild,
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
  },
})

export default memo(LoadingOverlay)
