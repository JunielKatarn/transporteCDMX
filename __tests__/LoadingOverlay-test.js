import React from 'react'
import { shallow } from 'enzyme'
import LoadingOverlay from '../src/components/LoadingOverlay'

describe('LoadingOverlay', () => {
  it('it renders correctly', () => {
    const wrapper = shallow(<LoadingOverlay showOverlay />)
    expect(wrapper).toMatchSnapshot()
  })

  it('it renders correctly when showOverlay is FALSE', () => {
    const wrapper = shallow(<LoadingOverlay />)
    expect(wrapper).toMatchSnapshot()
  })
})
