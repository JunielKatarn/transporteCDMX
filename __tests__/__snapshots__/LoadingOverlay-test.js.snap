// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay it renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
      },
      5,
    ]
  }
>
  <Memo(Loader)
    textStyle={
      Object {
        "color": "#FFF",
      }
    }
    visible={true}
  />
</View>
`;

exports[`LoadingOverlay it renders correctly when showOverlay is FALSE 1`] = `""`;
